pyrsmi/rocml.py

```diff
@@ -814,6 +1112,36 @@ def smi_get_device_unique_id(dev):
     """
     try:
         handle = _get_processor_handle(dev)
+        
+        # Use amdsmi package if available
+        if _using_amdsmi_package:
+            try:
+                import amdsmi
+                bdf_info = amdsmi.amdsmi_get_gpu_device_bdf(handle)
+                # bdf_info is a dict like {'domain': '0000', 'bus': 'c1', 'device': '00', 'function': '0'}
+                # or could be a BDF string like "0000:c1:00.0"
+                if isinstance(bdf_info, dict):
+                    domain = int(bdf_info.get('domain', '0'), 16)
+                    bus = int(bdf_info.get('bus', '0'), 16)
+                    device = int(bdf_info.get('device', '0'), 16)
+                    function = int(bdf_info.get('function', '0'), 16)
+                    # Encode as 64-bit: domain(32) | bus(8) | device(5) | function(3)
+                    return ((domain & 0xFFFFFFFF) << 32) | ((bus & 0xFF) << 8) | ((device & 0x1F) << 3) | (function & 0x7)
+                elif isinstance(bdf_info, str):
+                    # Parse BDF string like "0000:c1:00.0"
+                    parts = bdf_info.replace('.', ':').split(':')
+                    if len(parts) >= 4:
+                        domain = int(parts[0], 16)
```

Comment:
This is deeply nested in 2 try blocks and 3 conditionals, which can be hard to follow and debug. Can some of those layers be removed?